DOC-8738: Warn users that generating the debug zip can degrade the cluster (#17952)
cockroachdb/docs: src/current/v23.1/cockroach-debug-zip.md

### Use cases

{{site.data.alerts.callout_danger}}
 `cockroach debug zip` is an expensive operation and impacts cluster performance.

> The command is to be used only as an emergency measure under the guidance of Cockroach Labs.

Reviewer: ```suggestion
Only use this command as an emergency measure under the guidance of Cockroach Labs.
```

Author: updated

`--files-until` | End timestamp for log file, goroutine dump, and heap profile collection. This can be used to limit the size of the generated `.zip`, which is increased by these files. The timestamp uses the format `YYYY-MM-DD`, followed optionally by `HH:MM:SS` or `HH:MM`. For example:<br /><br />`--files-until='2021-07-01 16:00'`<br /><br />When specifying a narrow time window, we recommend adding extra seconds/minutes to account for uncertainties such as clock drift.<br /><br />**Default:** 24 hours beyond now (to include files created during `.zip` creation)
`--include-files` | [Files](#files) to include in the generated `.zip`. This can be used to limit the size of the generated `.zip`, and affects logs, heap profiles, goroutine dumps, and/or CPU profiles. The files are specified as a comma-separated list of [glob patterns](https://wikipedia.org/wiki/Glob_(programming)). For example:<br /><br />`--include-files=*.pprof`<br /><br />Note that this flag is applied _before_ `--exclude-files`. Use [`cockroach debug list-files`]({% link {{ page.version.version }}/cockroach-debug-list-files.md %}) with this flag to see a list of files that will be contained in the `.zip`.

> `--include-range-info` | Include information about ranges in <code>nodes/{node ID}/ranges.json</code> files, a single file per node.<br /><br />Only set this flag to `false` to exclude range information when all the following conditions are met:<ul><li>You are directed by Cockroach Labs support.</li><li>The issue under investigation is definitely not related to kv, for example, an error from a schema change about versions or flags or unsupported.</li><li>The cluster is large and you need to minimize the size of the `.zip` file.</ul>**Default:** true

Reviewer 2: I think for user-facing docs, I'd say be more specific than "related to kv". I might go with something like:

Include one file per node with information about the KV ranges stored on that node, in <code>nodes/{node ID}/ranges.json</code>. This information can be vital when debugging issues that involve the KV storage layer, such as data placement, load balancing, performance or other behaviors. In certain situations, on large clusters with large numbers of ranges, this file can be omitted if and only if the issue being investigated is already known to be in another layer of the system (for example an error message about an unsupported feature or incompatible value in a SQL schema change or statement). Note however many higher-level issues are ultimately related to the underlying KV storage layer described by this file so only set this to false if directed to do so by Cockroach Labs support. *Default:** true

Author: updated
